src/runtime/markdown-parser/handler/utils.ts

```diff
@@ -20,11 +20,13 @@ export function parseThematicBlock (lang: string) {
   const language = lang.replace(/[{|[](.+)/, '').match(/^[^ \t]+(?=[ \t]|$)/)
   const highlightTokens = lang.match(/{([^}]+)}/)
   const filenameTokens = lang.match(/\[(.+)\]/)
+  const meta = lang.replace(/^\w+\s*(\[[^\]]+\])?\s*(\{[^}]+\})?\s*/g, '')
```

Comment:
```suggestion
  const highlightTokens = lang.match(/{([^}]*)}/)
  const filenameTokens = lang.match(/\[([^\]]*)\]/)
  const meta = lang.replace(/^\w+\s*(\[[^\]]*\]|\{[^}]*\})?\s*(\[[^\]]*\]|\{[^}]*\})?\s*/, '')
```

This suggestion would allow empty `highlight` (`{}`) and `filename` (`[]`) blocks.
Therefore, if a user doesn't want to provide any of those, but use meta / wants to use the characters in the meta part, can do so.

It also makes the order of `highlight` and `filename` blocks arbitrary: `[]{}` and `{}[]` are both possible when replacing them for the meta.